selfservice/strategy/oidc/strategy.go

```diff
@@ -125,8 +125,7 @@ type authCodeContainer struct {
 }
 
 func generateState(flowID string) string {
-	state := x.NewUUID().String()
-	return base64.RawURLEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", flowID, state)))
+	return flowID
```

Comment:
@aeneasr / reviewers: this is potentially security relevant. Previously the state sent to the OIDC IdP was random, now it is "just" the flow ID.

The technical reason is that in the App/Code case, there is no shared state between the code that generates the state (for/in the app) and the code that invokes the OIDC callback (from the browser tab). So in the calllback we use the flow ID from the state to update the exchange with the session token.

In the OAuth flow, the state acts as a CSRF token to make sure that an attacker can't log in the user into his own account. However, here, the attacker would need to guess the victim's flow ID, which has high entropy.

Reply:
fixed